### PR TITLE
fix: max order greater than min order in ticketing system

### LINKED ISSUE #3521: Frontend check for min order and max order in ticketing system
**Describe the bug**
Their should be a frontend side check of minimum order and maximum order in ticketing system while creating any event

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'create an event'
2. fill the info
3. Add a ticketing system 
4. Minimum order value has no check of less than 1 and max order doesn't has check of greater than min order

**Expected behaviour**
it should be corrected

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: [MacOS]
 - Browser [ safari]

**Additional context**
I am taking this issue

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #3521 

#### Short description of what this resolves:
Added frontend check of minOrder and max order value in frontend

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by codedsun** [comment]
@kushthedude @shreyanshdwivedi Please check

**Comment by codedsun** [comment]
@iamareebjamal It has been implemented on server, also need to be implemented on client side

**Comment by codedsun** [comment]
@CosmicCoder96 @kushthedude Please review

**Comment by codedsun** [comment]
@kushthedude @iamareebjamal chck this

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -276,6 +276,10 @@ export default Component.extend(FormMixin, EventWizardMixin, {
             {
               type   : 'number',
               prompt : this.l10n.t('Invalid number')
+            },
+            {
```
Comment (kushthedude) [inline app/components/forms/wizard/basic-details-step.js]: Not required

File: app/mixins/event-wizard.js [hunk 2]
```diff
@@ -59,7 +59,14 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
...
     const numberOfTickets = data.tickets ? data.tickets.length : 0;
-    if (event.name && event.locationName && event.startsAtDate && event.endsAtDate && numberOfTickets > 0) {
+    let minMaxValueOfTicketValid = true;
+    for (const ticket of data.tickets ? data.tickets.toArray() : []) {
+      if (!(parseInt(ticket.get('maxOrder')) >= parseInt(ticket.get('minOrder')))) {
+        minMaxValueOfTicketValid = false;
+        break;
+      }
+    }
+    if (event.name && event.locationName && event.startsAtDate && event.endsAtDate && numberOfTickets > 0 && minMaxValueOfTicketValid) {
```
Comment (kushthedude) [inline app/mixins/event-wizard.js]: Not required in this way , as server will throw error error too for the same. If you can go in way like Semsntic UI validations, then the PR would make sense, as I will have to click submit button to fire up this check

Comment (codedsun) [reply app/mixins/event-wizard.js]: > I think the best place to add the check will be where we are getting data to create the event. Add a check just before hitting server. There are many checks implemented similarly.
Validation rules will not be helpful as you are using two different entities here.

It's mentioned in the issue

Comment (codedsun) [reply app/mixins/event-wizard.js]: Also, why required to hit the server, when the client side data is provided wrong? It makes sense that always the ticketMaxOrder will be greater than equal to ticketMinOrder We can add the check frontend side

File: app/components/forms/wizard/basic-details-step.js (hunk 1 above)
Comment (codedsun) [reply app/components/forms/wizard/basic-details-step.js]: @kushthedude why?

File: app/mixins/event-wizard.js (hunk 2 above)
Comment (codedsun) [reply app/mixins/event-wizard.js]: @kushthedude

**Comment by codedsun** [comment]
@kushthedude @iamareebjamal Check this

**Comment by kushthedude** [comment]
> I think the best place to add the check will be where we are getting data to create the event. Add a check just before hitting server. There are many checks implemented similarly.

If that's the case, then I don't think we need this check. Since we already have this check on the server. Also, This check only works when data is sent to the server but we indeed already have a similar check for this on the server. If this check can be implemented using `Semantic UI Validation` as has been implemented in case of donation ticket then it would be great to move forward.

**Comment by kushthedude** [comment]
> > I think the best place to add the check will be where we are getting data to create the event. Add a check just before hitting server. There are many checks implemented similarly.
> 
> If that's the case, then I don't think we need this check. Since we already have this check on the server. Also, This check only works when data is sent to the server but we indeed already have a similar check for this on the server. If this check can be implemented using `Semantic UI Validation` as has been implemented in case of donation ticket then it would be great to move forward.

@codedsun

**Comment by kushthedude** [comment]
Reopen when updated .

**Pull request closed without merging**